lib/ansible/plugins/connection/aws_ssm.py

```diff
@@ -359,7 +412,11 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
                 continue
 
             if mark_start in line:
-                begin = True
+                if self.is_windows:
+                    begin = True
+                else:
+                    if line.startswith("$") or line.strip() == mark_start:
+                        begin = True
```

Comment:
HI @j-shade, Observed these changes resulting the error mentioned at https://github.com/ansible/ansible/pull/49652#issuecomment-566598019. However, reverting back these changes to original, the integration test suite is passing for both linux and windows.

Reply:
Hi @hanumantharaomvl, sorry for the delay in getting back to you. Based on what i saw from your output, it looks as if your line starts with sh-4.2 instead of $, or the mark start hence it loops.

I have not experienced these issues when testing and debugging locally. If there are any further tests you wish me to run let me know.